process/rating/ratingsData.go

```diff
@@ -114,66 +117,100 @@ func NewRatingsData(args RatingsDataArg) (*RatingsData, error) {
 		metaRatingsStepData:         metaRatingStep,
 		shardRatingsStepData:        shardRatingStep,
 		selectionChances:            chances,
-		nodesSetup:                  args.NodesSetupHandler,
+		chainParametersHandler:      args.ChainParametersHolder,
 		ratingsSetup:                ratingsConfig,
 		roundDurationInMilliseconds: args.RoundDurationMilliseconds,
 	}
 
+	err = ratingData.computeRatingStepDataMaps(args.ChainParametersHolder.AllChainParameters())
+	if err != nil {
+		return nil, err
+	}
+
 	args.EpochNotifier.RegisterNotifyHandler(ratingData)
 
 	return ratingData, nil
 }
 
+func (rd *RatingsData) computeRatingStepDataMaps(chainParamsList []config.ChainParametersByEpochConfig) error {
+	shardRatingsSteps := make(map[uint32]process.RatingsStepHandler)
+	metaRatingsSteps := make(map[uint32]process.RatingsStepHandler)
+	for _, chainParams := range chainParamsList {
+		shardRatingsStepsArgs := computeRatingStepArg{
+			shardSize:                       chainParams.ShardMinNumNodes,
+			consensusSize:                   chainParams.ShardConsensusGroupSize,
+			roundTimeMillis:                 rd.roundDurationInMilliseconds,
+			startRating:                     rd.ratingsSetup.General.StartRating,
+			maxRating:                       rd.ratingsSetup.General.MaxRating,
+			hoursToMaxRatingFromStartRating: rd.ratingsSetup.ShardChain.HoursToMaxRatingFromStartRating,
+			proposerDecreaseFactor:          rd.ratingsSetup.ShardChain.ProposerDecreaseFactor,
+			validatorDecreaseFactor:         rd.ratingsSetup.ShardChain.ValidatorDecreaseFactor,
+			consecutiveMissedBlocksPenalty:  rd.ratingsSetup.ShardChain.ConsecutiveMissedBlocksPenalty,
+			proposerValidatorImportance:     rd.ratingsSetup.ShardChain.ProposerValidatorImportance,
+		}
+		shardRatingsStepData, err := computeRatingStep(shardRatingsStepsArgs)
+		if err != nil {
+			return fmt.Errorf("%w while computing shard rating steps for epoch %d", err, chainParams.EnableEpoch)
+		}
+		shardRatingsSteps[chainParams.EnableEpoch] = shardRatingsStepData
+
+		metaRatingsStepsArgs := computeRatingStepArg{
```

Comment:
kind of duplicated logic. However, can not see how to fix this as only 2 fields are the same between shardRatingsStepsArgs and metaRatingsStepsArgs